docs/basics/part-8a-using-earthly-in-your-current-ci.md

```diff
@@ -0,0 +1,93 @@
+
+```yaml
+# .github/workflows/ci.yml
+
+name: CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env:
+      DOCKERHUB_USERNAME: ${{ secrets.DOCKERHUB_USERNAME }}
+      DOCKERHUB_TOKEN: ${{ secrets.DOCKERHUB_TOKEN }}
+      FORCE_COLOR: 1
+    steps:
+    - uses: earthly/actions/setup-earthly@v1
+      with:
+        version: v0.7.0
+    - uses: actions/checkout@v2
+    - name: Put back the git branch into git (Earthly uses it for tagging)
+      run: |
+        branch=""
+        if [ -n "$GITHUB_HEAD_REF" ]; then
+          branch="$GITHUB_HEAD_REF"
+        else
+          branch="${GITHUB_REF##*/}"
+        fi
+        git checkout -b "$branch" || true
+    - name: Docker Login
+      run: docker login --username "$DOCKERHUB_USERNAME" --password "$DOCKERHUB_TOKEN"
+    - name: Run build
+      run: earthly --push +build
+```
+
+Here is an explanation of the steps above:
+
+* The action `earthly/actions/setup-earthly@v1` downloads and installs Earthly. Running this action is similar to running the Earthly installation one-liner `sudo /bin/sh -c 'wget https://github.com/earthly/earthly/releases/download/v0.6.30/earthly-linux-amd64 -O /usr/local/bin/earthly && chmod +x /usr/local/bin/earthly'`
+* The command `docker login` performs a login to the DockerHub registry. This is required, to prevent rate-limiting issues when using popular base images.
+* The command `earthly --ci --push +build` executes the build. The `--ci` flag is used here, in order to force the use of `--strict` mode -- in this mode, Earthly prevents the use of features that make the build less repeatable -- and also to disable local outputs. Artifacts and images resulting from the build are not needed within the CI environment. Any outputs should be pushed via `RUN --push` or `SAVE IMAGE --push` commands.
+
+### Using Earthly Satellites in CI (optional)
+
+Many CI systems are based on ephemeral sandboxes - meaning that when the build finishes, the environment is thrown away, and subsequent runs will have to start again from scratch, including downloading and reinstalling dependencies. This is the case for CIs such as GitHub Actions, GitLab CI, and CircleCI. This characteristic results in many steps being repeated on every run despite the fact that nothing related to those steps has changed.
+
+Earthly Satellites are a great way to mitigate this issue. As satellites have persistent cache that is instantly available, pipelines can execute much faster, when running together with the warmed-up cache of the remote instance. This alone typically results in a 2-20X speedup, depending on setup.
+
+To use a satellite in your CI, you need to:
+
+1. Generate an Earthly token and set it as an environment variable in your CI
+2. Specify the satellite name to use as part of the `earthly` command.
+
+To generate an Earthly token, assuming that you are logged in, you can use the command
```

Comment:
"To generate an Earthly token, assuming that you are logged in, you can use the command..."